Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/RuntimeOutputLog/RuntimeOutputLog.tsx

```diff
@@ -21,15 +33,73 @@ const createMarkup = (txt: string) => {
 };
 
 export const RuntimeOutputLog: React.FC<{ projectId: string }> = ({ projectId }) => {
-  const runtimeLogs = useRecoilValue(botRuntimeLogState(projectId));
-  const botRuntimeErrors = useRecoilValue(botRuntimeErrorState(projectId));
+  const runtimeData = useRecoilValue(runtimeStandardOutputDataState(projectId));
+  const botBuildErrors = useRecoilValue(botBuildTimeErrorState(projectId));
+  const { setRuntimeStandardOutputData, setApplicationLevelError } = useRecoilValue(dispatcherState);
+
   const runtimeLogsContainerRef = useRef<HTMLDivElement | null>(null);
 
+  const runtimeTrafficChannel = useRef<WebSocket | null>(null);
+
   useEffect(() => {
     if (runtimeLogsContainerRef?.current) {
       runtimeLogsContainerRef.current.scrollTop = runtimeLogsContainerRef.current.scrollHeight;
     }
-  }, [runtimeLogs, botRuntimeErrors]);
+  }, [runtimeData]);
+
+  useEffect(() => {
+    const setupLogConnection = async () => {
+      try {
+        const runtimeStreamUrl = await httpClient.get(`/publish/runtimeLogUrl/${projectId}`);
+
+        runtimeTrafficChannel.current = new WebSocket(runtimeStreamUrl.data);
+
+        if (runtimeTrafficChannel.current) {
+          runtimeTrafficChannel.current.onmessage = (event) => {
+            try {
```

Comment:
why do you need this try within the outer try, the catch block is doing the same thing for both